dali/operators/reader/loader/numpy_loader.cc

```diff
@@ -67,61 +135,12 @@ std::unique_ptr<FileStream> NumpyLoader::ParseHeader(std::unique_ptr<FileStream>
   nread = file->Read(token.data(), header_len);
   DALI_ENFORCE(nread == header_len, "Can not read header.");
   token[header_len] = '\0';
-  header = std::string(reinterpret_cast<char*>(token.data()));
-  DALI_ENFORCE(header.find_first_of("{") != std::string::npos, "Header is corrupted.");
+  header = std::string(reinterpret_cast<const char*>(token.data()));
+  DALI_ENFORCE(header.find("{") != std::string::npos, "Header is corrupted.");
   offset += header_len;
+  file->Seek(offset);  // prepare file for later reads
 
-  // prepare file for later reads
-  file->Seek(offset);
-
-  // extract dictionary info from header
-  std::smatch header_match;
-  DALI_ENFORCE(std::regex_search(header, header_match, header_regex_),
-               "Can not parse header.");
-
-  // now extract header information
-  // type
-  std::string typestring = header_match[1].str();
-
-  // < means LE, | means N/A, = means native. In all those cases, we can read
-  bool little_endian =
-    (typestring[0] == '<' || typestring[0] == '|' || typestring[0] == '=');
-  DALI_ENFORCE(little_endian,
-    "Big Endian files are not supported.");
-
-  std::string tid = typestring.substr(1);
-  // get type in a safe way
-  target.type_info = TypeFromNumpyStr(tid);
-
-  // check for data order
-  if (header_match[2].str() == "False")
-    target.fortran_order = false;
-  else
-    target.fortran_order = true;
-
-  // set sizes
-  std::string shapestring = header_match[3].str();
-  std::regex shape_regex{R"(,+)"};  // split on comma
-  std::sregex_token_iterator it{shapestring.begin(), shapestring.end(), shape_regex, -1};
-  std::vector<std::string> shapevec{it, {}};
-
-  // if shapevec size is 1 and shapevec[0] is the empty string,
-  // the array is actually a scalar/singleton (denoted as ())
-  // and thus the size needs to be set to one:
-  if ( (shapevec.size() == 1) && (shapevec[0] == "") ) shapevec[0] = "1";
-
-  // determine shapes
-  size_t shapesize = shapevec.size();
-  target.shape.resize(shapesize);
-  // cheapest thing to do is to define the tensor in an reversed way
-  if (target.fortran_order) {
-    for (size_t i = 0; i < shapesize; ++i)
-      target.shape[i] = static_cast<int64_t>(stoi(shapevec[shapesize-i-1]));
-  } else {
-    for (size_t i = 0; i < shapesize; ++i)
-      target.shape[i] = static_cast<int64_t>(stoi(shapevec[i]));
-  }
-
+  ParseHeaderMetadata(target, header);
```

Comment:
```suggestion
  ParseHeaderMetadata(target, std::move(header));
```
?

Reply:
not applicable any more (passed by reference now)